### PR TITLE
Add spawn progress watch modal, replace spawn_pending page

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
**Fixes**: 
Jira: https://issues.redhat.com/browse/ODH-513

**Analysis / Root cause**: 
As a DS user, I want to be able to track the progress of starting the NB server, so I can take appropriate action if there are any issues.

**Solution Description**: 
Replace the spawn-pending.html file with the spawn.html file and update the spawn application to start the server spawn process and display the progress in a dialog.

**Screen shots / Gifs for design review**: 
spawn-progress

/cc @dlabaj

### CODE REVIEW

**Comment by jeff-phillips-18** [comment]
@cdcabrera @dlabaj @kywalker-rh PTAL

**Review by kywalker-rh — APPROVED** [review]
lgtm

**Review by cdcabrera — APPROVED** [review]

File: jupyterhub_singleuser_profiles/ui/src/App/StartServerModal.tsx [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+  pageRef?: HTMLElement;
+};
+
+type SpawnStatus = {
+  status: 'success' | 'danger' | 'warning' | 'info' | 'default';
+  title: string;
+  reason: React.ReactNode;
+};
+
+const StartServerModal: React.FC<StartServerModalProps> = ({ shown, onClose, pageRef }) => {
```
Comment (dlabaj) [inline jupyterhub_singleuser_profiles/ui/src/App/StartServerModal.tsx:35]: Do we need an export here?

File: jupyterhub_singleuser_profiles/ui/src/App/StartServerModal.tsx [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+} from '@patternfly/react-core';
+import { APIGet } from '../utils/APICalls';
+import { HubUserRequest } from '../utils/HubCalls';
+import { useWatchSpawnProgress } from '../utils/useWatchSpawnProgress';
+import { CM_PATH, DEV_MODE, FOR_USER, USER } from '../utils/const';
+import { UserConfigMapType } from '../utils/types';
+
+import './StartServerModal.scss';
+
+type StartServerModalProps = {
```
Comment (dlabaj) [inline jupyterhub_singleuser_profiles/ui/src/App/StartServerModal.tsx:23]: Do we need to an export here as well?

**Review by dlabaj — COMMENTED** [review]
Looks good a couple of comments.

File: jupyterhub_singleuser_profiles/ui/src/App/StartServerModal.tsx (hunk 1 above)
Comment (jeff-phillips-18) [reply jupyterhub_singleuser_profiles/ui/src/App/StartServerModal.tsx:35]: It is exported as default

File: jupyterhub_singleuser_profiles/ui/src/App/StartServerModal.tsx (hunk 2 above)
Comment (jeff-phillips-18) [reply jupyterhub_singleuser_profiles/ui/src/App/StartServerModal.tsx:23]: Not at this time. Potentially if we add unit tests it _might_ be necessary but we can do that when we need to.

**Comment by lugi0** [comment]
Would it be possible to add an id to the progress bar element/div? e.g. `progress-bar` would be perfect for QE and ensure our test suite keeps working correctly with the new UI

**Review by dlabaj — APPROVED** [review]

**Review by dlabaj — APPROVED** [review]

**Pull request merged**